flink-connector-clickhouse/src/main/java/com/apache/flink/streaming/connectors/clickhouse/ClickHouseTableSink.java

```diff
@@ -0,0 +1,249 @@
+        this.address = address;
+        this.username = username;
+        this.password = password;
+        this.database = database;
+        this.table = table;
+        this.schema = schema;
+        this.batchSize = batchSize;
+        this.commitPadding = commitPadding;
+        this.retries = retries;
+        this.retryInterval = retryInterval;
+        this.ignoreInsertError = ignoreInsertError;
+
+    }
+
+    /**
+     * @param dataStream
+     * @deprecated
+     */
+    @Override
+    public void emitDataStream(DataStream<Row> dataStream) {
+        consumeDataStream(dataStream);
+    }
+
+    /**
+     *
+     * @return ClickHouseAppendSinkFunction
+     */
+    private ClickHouseAppendSinkFunction initSink() {
+        String prepareStatement = createPrepareStatement(schema, database, table);
+        return new ClickHouseAppendSinkFunction(address, username, password, prepareStatement, batchSize, commitPadding, retries, retryInterval, ignoreInsertError);
+    }
+
+    @Override
+    public TableSink<Row> configure(String[] strings, TypeInformation<?>[] typeInformations) {
+
+        ClickHouseTableSink copy;
+        try {
+            copy = new ClickHouseTableSink(address, username, password, database, table, schema, batchSize, commitPadding, retries, retryInterval, ignoreInsertError);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+
+        return copy;
+    }
+
+    @Override
+    public String[] getFieldNames() {
+        return schema.getFieldNames();
+    }
+
+    @Override
+    public TypeInformation<?>[] getFieldTypes() {
+        return schema.getFieldTypes();
+    }
+
+    @Override
+    public DataStreamSink<?> consumeDataStream(DataStream<Row> dataStream) {
+        return dataStream.addSink(initSink())
+                .setParallelism(dataStream.getParallelism())
+                .name(TableConnectorUtils.generateRuntimeName(this.getClass(), schema.getFieldNames()));
+    }
+
+    @Override
+    public DataType getConsumedDataType() {
+        return schema.toRowDataType();
+    }
+
+
+    @Override
+    public TypeInformation<Row> getOutputType() {
+        return new RowTypeInfo(schema.getFieldTypes(), schema.getFieldNames());
+    }
+
+
+    @Override
+    public TableSchema getTableSchema() {
+        return schema;
+    }
+
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+
+    public String createPrepareStatement(TableSchema tableSchema, String database, String table) {
+        String[] fieldNames = tableSchema.getFieldNames();
+        String columns = String.join(",", fieldNames);
+        String[] questionMark = new String[fieldNames.length];
+        for (int i = 0; i < questionMark.length; i++) {
```

Comment:
this could be change to a stream map no?

String questionMark = Arrays.stream(fieldNames)
                .map(field -> "?")
                .reduce((left,right) -> left+","+right)
                .get();